src/github.com/matrix-org/dendrite/common/config/config.go

```diff
@@ -360,6 +375,15 @@ func (config *Dendrite) derive() {
 		config.Derived.Registration.Flows = append(config.Derived.Registration.Flows,
 			authtypes.Flow{Stages: []authtypes.LoginType{authtypes.LoginTypeDummy}})
 	}
+
+	// Load application service configuration files
+	if len(config.ApplicationService.ConfigFiles) > 0 {
+		if err := loadAppservices(config); err != nil {
+			return err
+		}
+	}
```

Comment:
I wouldn't bother putting the non-empty guard there, the `loadAppservices` function is happy with getting a config without any appservices I think.